tests/dragonfly/redis_replicaiton_test.py

```diff
@@ -37,43 +66,126 @@ def redis_server() -> RedisServer:
+async def test_replication_stable_sync(df_local_factory, df_seeder_factory, redis_server, t_replicas, seeder_config):
+    master = redis_server
+    c_master = aioredis.Redis(port=master.port)
     assert await c_master.ping()
 
-    replica = df_local_factory.create(port=redis_server.port + 1)
+    replica = df_local_factory.create(port=master.port + 1, proactor_threads=t_replicas[0])
     replica.start()
     c_replica = aioredis.Redis(port=replica.port)
     assert await c_replica.ping()
 
-    await c_replica.execute_command("REPLICAOF", "localhost", redis_server.port)
+    await c_replica.execute_command("REPLICAOF", "localhost", master.port)
     await wait_available_async(c_replica)
 
-    seeder = df_seeder_factory.create(port=redis_server.port, keys=100, dbcount=1, unsupported_types=[ValueType.JSON])
-    await seeder.run(target_ops=100)
+    seeder = df_seeder_factory.create(port=master.port, **seeder_config)
+    await seeder.run(target_ops=1000)
+
+    await await_synced(master.port, replica.port, seeder_config["dbcount"])
 
-    await asyncio.sleep(5)
     capture = await seeder.capture()
     assert await seeder.compare(capture, port=replica.port)
+
+
+# Threads for each dragonfly replica, Seeder Config.
+replication_specs = [
+    ([1], dict(keys=1000, dbcount=1, unsupported_types=[ValueType.JSON])),
+    # ([6, 6, 6], dict(keys=4_000, dbcount=4, unsupported_types=[ValueType.JSON])),
+    # ([2, 2, 2, 2], dict(keys=4_000, dbcount=4, unsupported_types=[ValueType.JSON])),
+    # ([8, 8], dict(keys=4_000, dbcount=4, unsupported_types=[ValueType.JSON])),
+    # ([1] * 8, dict(keys=500, dbcount=2, unsupported_types=[ValueType.JSON])),
+    # ([1], dict(keys=100, dbcount=2, unsupported_types=[ValueType.JSON])),
+]
+@pytest.mark.asyncio
+@pytest.mark.parametrize("t_replicas, seeder_config", replication_specs)
+@pytest.mark.skip(reason="Skipping until we fix replication from redis")
+async def test_redis_replication_all(df_local_factory, df_seeder_factory, redis_server, t_replicas, seeder_config):
+    master = redis_server
+    c_master = aioredis.Redis(port=master.port)
+    assert await c_master.ping()
+
+    replicas = [
+        df_local_factory.create(port=master.port+i+1, proactor_threads=t)
+        for i, t in enumerate(t_replicas)
+    ]
+
+    # Fill master with test data
+    seeder = df_seeder_factory.create(port=master.port, **seeder_config)
+    await seeder.run(target_deviation=0.1)
+
+    # Start replicas
+    df_local_factory.start_all(replicas)
+
+    c_replicas = [aioredis.Redis(port=replica.port) for replica in replicas]
+
+    # Start data stream
+    stream_task = asyncio.create_task(seeder.run(target_ops=6000))
+    await asyncio.sleep(0.0)
+
+    # Start replication
+    async def run_replication(c_replica):
+        await c_replica.execute_command("REPLICAOF localhost " + str(master.port))
+        await wait_available_async(c_replica)
```

Comment:
when you are calling wait_available_async inside this loop this means we always have only one replica in the load phase, is this intended?